File: ios/BlakJaks/Shared/Components/NicotineWarningBanner.swift
import SwiftUI

// MARK: - NicotineWarningBanner
// FDA-compliant nicotine warning per 21 CFR § 1143.3(b)(2).
// Required on: iOS splash screen, Shop, Cart, Checkout.
// NOT shown on: Insights, Social, Wallet, Scanner, Profile.
//
// Spec:
// - Occupies 20% of screen height (top)
// - Black background (#000000)
// - White text (#FFFFFF)
// - Helvetica Bold / Arial Bold font (FDA-specified)
// - Text dynamically sized to fill banner area without overflow
// - NEVER dismissible — no close button
//
// Usage: Add to top of required views, offset content by 20% screen height.
// Pages that include this MUST add .padding(.top, bannerHeight) to content below.

struct NicotineWarningBanner: View {
    private let warningText = "WARNING: This product contains nicotine. Nicotine is an addictive chemical."

    var body: some View {
        GeometryReader { geo in
            let bannerHeight = geo.size.height * 0.20

            ZStack {
                Color.black

                Text(warningText)
                    .font(.custom("Helvetica-Bold", size: dynamicFontSize(for: geo.size)))
                    .foregroundColor(.white)
                    .multilineTextAlignment(.center)
                    .minimumScaleFactor(0.5)
                    .padding(.horizontal, Spacing.md)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(height: bannerHeight)
            .frame(maxWidth: .infinity)
        }
        .frame(height: UIScreen.main.bounds.height * 0.20)
        .ignoresSafeArea(edges: .top)
    }

    // Dynamically size text to fill the banner without overflow
    private func dynamicFontSize(for size: CGSize) -> CGFloat {
        let bannerHeight = size.height * 0.20
        // Start large, let minimumScaleFactor handle reduction
        return min(bannerHeight * 0.22, 24)
    }
}

// MARK: - View modifier for pages that require the banner

struct NicotineWarningBannerModifier: ViewModifier {
    func body(content: Content) -> some View {
        let bannerHeight = UIScreen.main.bounds.height * 0.20
        ZStack(alignment: .top) {
            content
                .padding(.top, bannerHeight)
            NicotineWarningBanner()
        }
    }
}

extension View {
    /// Adds the FDA-required nicotine warning banner to the top of this view.
    /// Content is automatically offset by 20% screen height.
    /// Only apply to: Splash/Welcome, Shop, Cart, Checkout.
    func withNicotineWarning() -> some View {
        modifier(NicotineWarningBannerModifier())
    }
}

// MARK: - Preview

#Preview {
    VStack(spacing: 0) {
        NicotineWarningBanner()
        ScrollView {
            VStack {
                Text("Shop Content")
                    .font(.largeTitle)
                    .padding(.top, 20)
            }
        }
    }
    .background(Color.backgroundPrimary)
}

File: android/app/src/main/java/com/blakjaks/app/core/components/NicotineWarningBanner.kt
package com.blakjaks.app.core.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// FDA-required nicotine warning. Never dismissible.
// Add ONLY to: SplashScreen, ShopScreen, CartScreen, CheckoutScreen.
@Composable
fun NicotineWarningBanner(modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .heightIn(min = (0.20f * 800).dp) // ~20% of screen
            .background(Color.Black),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = "WARNING: This product contains nicotine. Nicotine is an addictive chemical.",
            color = Color.White,
            fontSize = 14.sp,
            fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(horizontal = 24.dp, vertical = 16.dp)
        )
    }
}
